app/src/main/java/org/jellyfin/androidtv/ui/home/HomeFragmentNowPlayingRow.kt

```diff
@@ -33,7 +33,10 @@ class HomeFragmentNowPlayingRow(
 	}
 
 	fun update(rowsAdapter: ArrayObjectAdapter) {
-		if (mediaManager.isPlayingAudio) add(rowsAdapter)
+		if (mediaManager.isPlayingAudio) {
+			remove(rowsAdapter)
+			add(rowsAdapter)
```

Comment:
I think this should use the update function in the adapter

https://developer.android.com/reference/androidx/leanback/widget/ArrayObjectAdapter#replace

Reply:
done. I'm not super familiar with the nuances of garbage collection in java. I hope just reassigning `row` to a new `ListRow` doesn't cause a leak.

37: `if (row != null) {`
38: `row = ListRow(HeaderItem(context.getString(R.string.lbl_now_playing)), mediaManager.managedAudioQueue)`
39: `rowsAdapter.replace(0, row)`